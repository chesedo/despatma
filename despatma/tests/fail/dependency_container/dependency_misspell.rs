struct Config {
    port: u32,
}

struct Service;

impl Service {
    fn new(port: u32) -> Self {
        println!("Service started on port {}", port);
        Self
    }
}

#[despatma::dependency_container]
impl DependencyContainer {
    fn config(&self) -> Config {
        Config { port: 8080 }
    }

    fn service(&self, canfi: Config) -> Service {
        Service::new(config.port)
    }
}
