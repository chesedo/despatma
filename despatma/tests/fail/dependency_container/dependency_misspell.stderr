error: The 'canfi' dependency has not been registered

         = help: Did you mean `config`?

  --> tests/fail/dependency_container/dependency_misspell.rs:20:23
   |
20 |     fn service(&self, canfi: Config) -> Service {
   |                       ^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/fail/dependency_container/dependency_misspell.rs:23:2
   |
23 | }
   |  ^ consider adding a `main` function to `$DIR/tests/fail/dependency_container/dependency_misspell.rs`
